addons/hr_payroll/models/hr_benefit.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import pytz
+import itertools
+from psycopg2 import IntegrityError
+from dateutil.relativedelta import relativedelta
+from odoo import api, fields, models, exceptions, _
+from odoo.tools import mute_logger
+from odoo.exceptions import UserError, ValidationError
+from odoo.addons.resource.models.resource import Intervals
+
+class HrBenefit(models.Model):
+    _name = 'hr.benefit'
+    _description = 'hr.benefit'
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(default=True)
+    employee_id = fields.Many2one('hr.employee', required=True,
+        domain=lambda self: [('contract_ids.state', 'in', ('open', 'pending')), ('company_id', '=', self.env.user.company_id.id)])
```

Comment:
There are no multi company record rule on employees. But we can removed it.

Reply:
Wouldn't it make sense to have this record rule? Was it removed on purpose?
Additionally, I assume that most "hr.benefit" records will be created automatically, and HR Officers will mostly review/edit them, right? If that's correct, then selecting a different employee won't be a common use case, and the domain is a bit useless.

Reply:
You are right, the main use case for HR is reviewing `hr.benefit`.
On the other hand, if one benefit is manually created with an employee from another company, it will affect payslips of the other company which we don't want. We should add multi company record rules either on `hr.employee` or at least on `hr.benefit` to avoid that.

Reply:
It was removed on purpose: "Someone" would like to see all the employees to make stats. 

I've already created several feedbacks to reintroduce them, always refused.

We should add an ir.rule on benefits (Mostly for privacy and to avoid generating payslips for indian people, ...).

Reply:
> It was removed on purpose: "Someone" would like to see all the employees to make stats.

We have general purpose multi-company rules everywhere, this one should be added for consistency. Customising record rules for specific companies is something that is done by the admin, on a case by case basis.
@Musvol If there is a real risk of messing up data, the sanity check should be done in the business code before we perform the actual action, rather than hoping all is well because there was a domain on the employee_id (employees could theoretically be reassigned to other companies, etc.).